liveblog/liveblog
# [LBSD-2733] - Introduce custom Notify module for Improved UI synchronization

### Purpose

This overhauls the existing notification system (taken from [superdesk-client-core/notify](https://github.com/liveblog/superdesk-client-core/blob/51e26f399689aa0d739d0acdb058bff7b6618644/scripts/core/notify/notify.js)), introducing a refactored notify module that addresses critical issues with UI synchronization and extends the module's capabilities for future enhancements. 

The main reason for the introduction of this custom `Notify` module is because the original service that we were using from `superdesk-client-core` lacks a proper way to remove messages. The `pop` method tends to remove the wrong message as it simply removes the last element in the array. This could be observed for messages that we want to keep some for longer times and suddenly a pop call would remove them unexpectedly.

The primary goal of this refactor is to ensure that notification messages are managed in a way that keeps the UI consistently updated in response to changes, specifically message removals. 

### What has changed

- Introduction of Unique Message IDs: Implemented unique IDs for each message to facilitate precise message management. This allows for the specific removal of messages without relying on array order

- Simplified Message Addition and Removal: Streamlined the process of adding and removing messages by encapsulating the functionality into more concise and reusable methods
 
- TypeScript Refactoring for Type Safety and Clarity: Migrated the module from JavaScript to TypeScript

- Updated references to the service and remove the use of `pop` approach and instead use `removeById`

- Converted constants from vanilla js into typescript module

### Steps to test

- Checkout the branch and run the services
- Click around and the see the notifications working as expected


Part of [LBSD-2733]

[LBSD-2733]: https://sofab.atlassian.net/browse/LBSD-2733?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
